### PR TITLE
V14 added acceptance tests for the Stylesheets

### PR DESCRIPTION
**Pull request by nhudinh0309**
### To test
Follow the README.md located in tests/Umbraco.Tests.AcceptanceTest for running the test
Test the tests in tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Scripts.spec.ts

### Description
Updated API test for the stylesheet to work with the updated testHelpers
Added Stylesheet UI acceptance test

### CODE REVIEW

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+  test('can create a stylesheet file', async ({umbracoApi, umbracoUi}) => {
+    // Arrange
+    await umbracoApi.stylesheet.ensureNameNotExists(styleSheetFileName);
+
+    //Act
+    await umbracoUi.stylesheet.openActionsMenuAtRoot();
+    await umbracoUi.stylesheet.clickNewStylesheetFileButton();
+    await umbracoUi.stylesheet.enterStylesheetName(styleSheetName);
+    await umbracoUi.stylesheet.clickSaveButton();
```
Comment (andr317c) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts:26]:
```suggestion
    await umbracoUi.waitForTimeout(1000);
    await umbracoUi.stylesheet.clickSaveButton();
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    // TODO: when frontend is ready, verify the removed folder is NOT displayed under the Stylesheets section
+    // TODO: when frontend is ready, verify the notification displays
+  });
+
+  // TODO: remove skip when frontend is ready as currently it will create 2 child folders in UI when creating a folder in a folder
+  test.skip('can create a folder in a folder', async ({umbracoApi, umbracoUi}) => {
+    // Arrange
+    await umbracoApi.stylesheet.ensureNameNotExists(styleFolderName);
+    await umbracoApi.stylesheet.createFolder(styleFolderName);
+    const childsFolderName = "ChildFolderName";
```
Comment (andr317c) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts:130]:
```suggestion
    const childFolderName = "ChildFolderName";
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  test('can create a new Rich Text Editor style sheet file', async ({umbracoApi, umbracoUi}) => {
+    // Arrange
+    await umbracoApi.stylesheet.ensureNameNotExists(styleSheetFileName);
+
+    //Act
+    await umbracoUi.stylesheet.openActionsMenuAtRoot();
+    await umbracoUi.stylesheet.clickNewRTEStylesheetFileButton();
+    await umbracoUi.stylesheet.enterStylesheetName(styleSheetName);
+    await umbracoUi.stylesheet.addNewRule(ruleName, 'h1', 'color:red');
+    await umbracoUi.stylesheet.clickSaveButton();
```
Comment (andr317c) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts:46]:
```suggestion
    await umbracoUi.waitForTimeout(1000);
    await umbracoUi.stylesheet.clickSaveButton();
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  test.skip('can create a folder in a folder', async ({umbracoApi, umbracoUi}) => {
+    // Arrange
+    await umbracoApi.stylesheet.ensureNameNotExists(styleFolderName);
+    await umbracoApi.stylesheet.createFolder(styleFolderName);
+    const childsFolderName = "ChildFolderName";
+
+    // Act
+    await umbracoUi.stylesheet.clickRootFolderCaretButton();
+    await umbracoUi.stylesheet.openActionsMenuForName(styleFolderName);
+    await umbracoUi.stylesheet.createNewFolder(childsFolderName);
```
Comment (andr317c) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts:135]:
```suggestion
    await umbracoUi.stylesheet.createNewFolder(childFolderName);
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+    //Assert 
+    expect(await umbracoApi.stylesheet.doesNameExist(childsFolderName)).toBeTruthy();
```
Comment (andr317c) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts:138]:
```suggestion
    expect(await umbracoApi.stylesheet.doesNameExist(childFolderName)).toBeTruthy();
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    const styleChildren = await umbracoApi.stylesheet.getChildren(styleFolderName);
+    expect(styleChildren[0].path).toBe(styleFolderName + '/' + childsFolderName);
```
Comment (andr317c) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts:140]:
```suggestion
    expect(styleChildren[0].path).toBe(styleFolderName + '/' + childsFoldeName);
```

**Review by andr317c — CHANGES_REQUESTED** [review]

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts (hunk 1 above)
Comment (nhudinh0309) [reply tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts:24]: waitForTimeout is unnecessary here as the test still works fine without it

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts (hunk 3 above)
Comment (nhudinh0309) [reply tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts:44]: waitForTimeout is unnecessary here as the test still works fine without it

File: tests/Umbraco.Tests.AcceptanceTest/package-lock.json
[the remark this answers is not in the document]
Comment (nhudinh0309) [reply tests/Umbraco.Tests.AcceptanceTest/package-lock.json:1]: ok, I get it

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts (hunk 1 above)
Comment (andr317c) [reply tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts:24]: This test is failing for me when I run it 🙁

**Comment by JasonElkin** [comment]
@nhudinh0309 you can ignore this comment.

Just commenting to remove it from the core collabs dashboard, and flagging to @nul800sebastiaan that this blue-on-blue pull request is triggering the contributors actions 🤔

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts (hunk 1 above)
Comment (nhudinh0309) [reply tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/Stylesheets.spec.ts:26]: Ok, we add it for now and when frontend validation is implemented, we will remove it

**Review by andr317c — APPROVED** [review]
LGTM

**Pull request merged**